controllers/tenantcontrolplane_controller.go

```diff
@@ -84,16 +84,15 @@ func (r *TenantControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 	defer cancelFn()
 
 	tenantControlPlane, err := r.getTenantControlPlane(ctx, req.NamespacedName)()
-	if err != nil {
-		if apimachineryerrors.IsNotFound(err) {
-			log.Info("resource may have been deleted, skipping")
-
-			return ctrl.Result{}, nil
-		}
+	if k8serrors.IsNotFound(err) {
```

Comment:
I think a check here is required to avoid a nil pointer dereference.

Reply:
It's not necessary, we don't have to nest it: `IsNotFound returns true if the specified error was created by NewNotFound. It supports wrapped errors and returns false when the error is nil.`

Reply:
TIL, thanks!